definitions/sources/src_bmg_all_waves.sqlx

```diff
@@ -1,16 +1,13 @@
 config {
-    type: "view",
-    dependencies: ["constructor_src_bmg_all_waves"]
+  type: "view",
 }
 
-  -- This is a "declaration" file. Its purpose is to make the view created by our
-  -- operations script available to the rest of the Dataform project via ref().
-  -- The SQL here is a simple self-reference. It acts as a check to ensure
-  -- the view was actually created by the operations script. If the view doesn't exist,
-  -- this compilation will fail, preventing downstream errors. 
-  -- I could also assign a js variable src_bmg_all_waves and use it upstream in 
-  -- create_template_src_bmg_waves_union and use self reference such as SELECT * FROM ${self()} to check table creation
-SELECT
-  * 
-FROM
-  ${ref("constructor_src_bmg_all_waves")}
+SELECT 
+    LOWER(REGEXP_REPLACE(
+        REGEXP_REPLACE(_TABLE_SUFFIX, r'_', ' '),  -- Replace underscores with spaces
+        r'(\d+)', 
+        r'wave \1'  -- Add "wave " before numbers
+    )) AS wave_name,
+    *
+FROM `govuk-polling.govuk_polling_responses.src_bmg_wave_*`
```

Comment:
Just a small point - `FROM `govuk-polling.govuk_polling_responses.src_bmg_*` would allow us to get the "wave_x" as table name and not have to add that on.

Reply:
Maybe we could clean out that dataset as part of a pairing session?